lit/docs/operation/creds/kubernetes.lit

```diff
@@ -0,0 +1,251 @@
+CONCOURSE_KUBERNETES_CONFIG_PATH=~/.kube/config
+}}}
+
+
+\section{
+  \title{Credential lookup rules}{kubernetes-credential-lookup-rules}
+
+  When resolving a parameter such as \code{((foo))}, Concourse will lookup for
+  it in the following order in the namespace configured for that team:
+
+  \list{
+    \codeblock{c}{{{
+    Name:         PIPELINE_NAME.foo
+    Namespace:    concourse-TEAM_NAME
+    Type:         Opaque
+
+    Data
+    ====
+    value:        32 bytes
+    }}}
+  }{
+    \codeblock{c}{{{
+    Name:         foo
+    Namespace:    concourse-TEAM_NAME
+    Type:         Opaque
+
+    Data
+    ====
+    value:        32 bytes
+    }}}
+  }
+
+  You can also have nested fields, which can be accessed using \code{.} syntax,
+  e.g. \code{((foo.bar))}:
+
+  \list{
+    \codeblock{c}{{{
+    Name:         PIPELINE_NAME.foo
+    Namespace:    concourse-TEAM_NAME
+    Type:         Opaque
+
+    Data
+    ====
+    bar:          32 bytes
+    }}}
+  }{
+    \codeblock{c}{{{
+    Name:         foo
+    Namespace:    concourse-TEAM_NAME
+    Type:         Opaque
+
+    Data
+    ====
+    bar:          32 bytes
+    }}}
+  }
+
+  The prefix prepended to the namespace used by Concourse to search for secrets
+  (in the examples above, \code{concourse-}) can be changed by configuring the
+  following in the \reference{web-node}:
+
+  \codeblock{bash}{{{
+  CONCOURSE_KUBERNETES_NAMESPACE_PREFIX=some-other-prefix-
+  }}}
+
+
+  If an action is being run in a one-off build, Concourse will not include the
+  pipeline name in the secret that it looks for.
+
+}
+
+\section{
+  \title{Configuring Kubernetes RBAC}
+
+  As the Web nodes need to retrieve secrets from namespaces that are not its
+  own, it needs extra permissions to do so.
+
+  If you have \link{RBAC}{https://kubernetes.io/docs/reference/access-authn-authz/rbac/}
+  enabled, that means creating the necessary Kubernetes objects to identify the
+  Web nodes and give them access to a predefined list of namespaces where the
+  secrets live.
+
+  Regardless of how the Kubernetes RBAC-related objects are created, the basic
+  requirement is that \bold{\code{web} must be able to read secrets in the 
+  namespaces where each teams' secrets live}.
+
+  For instance, if you have the following teams which you want to read secrets
+  from:
+
+	\list{
+		team-a
+	}{
+		team-b
+	}
+
+  Assuming the following \reference{web-node} configuration:
+
+  \codeblock{bash}{{{
+  CONCOURSE_KUBERNETES_NAMESPACE_PREFIX=myprefix-
+  }}}
+
+  \code{web} must be able to \code{get} secrets from the following namespaces:
+
+
+	\list{
+		myprefix-team-a
+	}{
+		myprefix-team-b
+	}
+
+  As an example, the following sample RBAC Kubernetes could be created to allow
+  \code{web} to interpolate credentials for "team-a" and "team-b":
+
+
+  \codeblock{yaml}{{{
+  ---
+  # Identifies the `web` service as an actor.
+  #
+  # ref: https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/
```

Comment:
@vito do you think this kind of comment is too much? Providing references to the document from the official Kubernetes website?

Reply:
Nah that's super useful. :+1: